L1Trigger/L1THGCal/interface/concentrator/HGCalConcentratorSuperTriggerCellImpl.h

```
@@ -23,53 +23,75 @@ class HGCalConcentratorSuperTriggerCellImpl
 
+    enum EnergyDivisionType{
+      superTriggerCell,
+      oneBitFraction,
+      equalShare,
+    };
+    EnergyDivisionType energyDivisionType_;
+    static constexpr int kNLayers_ = 3;
     HGCalTriggerTools triggerTools_;
-    HGCSiliconDetIdToROC detIdToROC_;
+    HGCalCoarseTriggerCellMapping coarseTCmapping_;
     std::vector<unsigned> stcSize_;
+    bool fixedDataSizePerHGCROC_;
+
+    //Parameters for energyDivisionType_ = oneBitFraction
+    double oneBitFractionThreshold_;
+    double oneBitFractionLowValue_;
+    double oneBitFractionHighValue_;
+
+    //Parameters for energyDivisionType_ = equalShare
+    int nTriggerCellsForDivision_;
 
     class SuperTriggerCell {
   
     private:
-        float sumPt_, sumMipPt_;
-        int sumHwPt_, maxHwPt_; 
+        float sumPt_, sumMipPt_, fracsum_;
+        int sumHwPt_, maxHwPt_, stcId_; 
         unsigned maxId_;
-        
+
     public:
-        SuperTriggerCell(){  sumPt_=0, sumMipPt_=0, sumHwPt_=0, maxHwPt_=0, maxId_=0 ;}
-        void add(const l1t::HGCalTriggerCell &c) {
+        SuperTriggerCell(){  
+          sumPt_=0, sumMipPt_=0, sumHwPt_=0, maxHwPt_=0, maxId_=0, fracsum_ = 0, stcId_ = 0; }
+        void add(const l1t::HGCalTriggerCell &c, int stcId) {
             sumPt_ += c.pt();
             sumMipPt_ += c.mipPt();
             sumHwPt_ += c.hwPt();
             if (maxId_ == 0 || c.hwPt() > maxHwPt_) {
                 maxHwPt_ = c.hwPt();
                 maxId_ = c.detId();
             }
+            
+            if  ( stcId_ == 0 ){
+              stcId_ = stcId;
+            }
+
         }
-        void assignEnergy(l1t::HGCalTriggerCell &c) const {
-            c.setHwPt(sumHwPt_);
-            c.setMipPt(sumMipPt_);
-            c.setPt( sumPt_ );
+        void SetMaxTC(const l1t::HGCalTriggerCell &c){
+          maxId_ = c.detId();
+          maxHwPt_ = c.hwPt();
         }
+	void addToFractionSum(float frac){
+	  fracsum_ += frac;
+	  if ( fracsum_ > 1 ) fracsum_ = 1;
```

If this shouldn't happen, it might be better to throw an exception.